it/src/main/java/com/google/cloud/teleport/it/matchers/RecordsSubject.java

```diff
@@ -172,4 +192,172 @@ public void allMatch(Map<String, Object> record) {
+        records.add(converted);
+      }
+
+      return records;
+    } catch (Exception e) {
+      throw new RuntimeException("Error converting TableResult to Records", e);
+    }
+  }
+
+  /**
+   * Convert Spanner {@link Struct} list to a list of maps.
+   *
+   * @param structs Structs to parse
+   * @return List of maps to use in {@link RecordsSubject}
+   */
+  public static List<Map<String, Object>> structsToRecords(List<Struct> structs) {
+    try {
+      List<Map<String, Object>> records = new ArrayList<>();
+
+      for (Struct struct : structs) {
+        Map<String, Object> record = new HashMap<>();
+
+        for (Type.StructField field : struct.getType().getStructFields()) {
+          Value fieldValue = struct.getValue(field.getName());
+          // May need to explore using typed methods instead of .toString()
+          record.put(field.getName(), fieldValue.toString());
+        }
+
+        records.add(record);
+      }
+
+      return records;
+    } catch (Exception e) {
+      throw new RuntimeException("Error converting TableResult to Records", e);
+    }
+  }
+
+  /**
+   * Convert Avro {@link GenericRecord} to a list of maps.
+   *
+   * @param avroRecords Avro Records to parse
+   * @return List of maps to use in {@link RecordsSubject}
+   */
+  public static List<Map<String, Object>> genericRecordToRecords(List<GenericRecord> avroRecords) {
+    try {
+      List<Map<String, Object>> records = new ArrayList<>();
+
+      for (GenericRecord row : avroRecords) {
+        Map<String, Object> converted = objectMapper.readValue(row.toString(), recordTypeReference);
+        records.add(converted);
+      }
+
+      return records;
+    } catch (Exception e) {
+      throw new RuntimeException("Error converting Avro Record to Map", e);
+    }
+  }
+
+  /**
+   * Convert BigQuery {@link InsertAllRequest.RowToInsert} to a list of maps.
+   *
+   * @param rows BigQuery rows to parse.
+   * @return List of maps to use in {@link RecordsSubject}
+   */
+  public static List<Map<String, Object>> bigQueryRowsToRecords(
+      List<InsertAllRequest.RowToInsert> rows) {
+    try {
+      List<Map<String, Object>> records = new ArrayList<>();
+      rows.forEach(row -> records.add(new HashMap<>(row.getContent())));
+
+      return records;
+    } catch (Exception e) {
+      throw new RuntimeException("Error converting BigQuery Row to Map", e);
+    }
+  }
+
+  /**
+   * Convert BigQuery {@link InsertAllRequest.RowToInsert} to a list of maps.
+   *
+   * @param rows BigQuery rows to parse.
+   * @param excludeCols BigQuery rows to filter out of result.
+   * @return List of maps to use in {@link RecordsSubject}
+   */
+  public static List<Map<String, Object>> bigQueryRowsToRecords(
+      List<InsertAllRequest.RowToInsert> rows, List<String> excludeCols) {
+    List<Map<String, Object>> records = bigQueryRowsToRecords(rows);
+    try {
+      excludeCols.forEach(col -> records.forEach(row -> row.remove(col)));
```

Comment:
could be a separate method

Reply:
I will separate this out in a future PR.

Reply:
I will separate this out in a future PR.